CollapseLauncher/Classes/InstallManagement/BaseClass/InstallManagerBase.cs

```diff
@@ -481,17 +483,168 @@ public async Task MoveGameLocation()
+                                filesToDelete = new string[] { "ACE-BASE.sys", "GameAssembly.dll", "pkg_version", "config.ini", "^StarRail.*", "^Unity.*" };
+                                foldersToKeepInData = new string[] { "ScreenShots" };
+                                break;
+
+                            case GameType.Genshin:
+                                switch (ConfigV2Store.CurrentConfigV2GameRegion)
+                                {
+                                    case "Global":
+                                        gameDataFolderName = "GenshinImpact_Data";
+                                        foldersToDelete = new string[] { "GenshinImpact_Data" };
+                                        filesToDelete = new string[] { "HoYoKProtect.sys", "pkg_version", "GenshinImpact.exe", "UnityPlayer.dll", "config.ini", "^mhyp.*", "^Audio.*" };
+                                        foldersToKeepInData = null;
+                                        break;
+
+                                    case "Mainland China":
+                                        gameDataFolderName = "YuanShen_Data";
+                                        foldersToDelete = new string[] { "YuanShen_Data" };
+                                        filesToDelete = new string[] { "HoYoKProtect.sys", "pkg_version", "YuanShen.exe", "UnityPlayer.dll", "config.ini", "^mhyp.*", "^Audio.*" };
+                                        foldersToKeepInData = null;
+                                        break;
+
+                                    default:
+                                        LogWriteLine($"Unknown GI Game Region! : {ConfigV2Store.CurrentConfigV2GameRegion}", LogType.Error, true);
+                                        break;
+                                }
+                                break;
+
+                            case GameType.Zenless:
+                                LogWriteLine($"Cannot uninstall game: Zenless uninstall method is not yet implemented!", LogType.Error, true);
+                                return false;
+
+                            default:
+                                LogWriteLine($"Cannot uninstall game: GameType is Unknown!", LogType.Error, true);
+                                return false;
+                        }
+
+                        // Cleanup Game_Data folder while keeping whatever specified in foldersToKeepInData
+                        foreach (string folderGameData in Directory.GetDirectories(Path.Combine(GameFolder, gameDataFolderName)))
+                        {
+                            try
+                            {
+                                if (foldersToKeepInData != null && !foldersToKeepInData.Contains(folderGameData)) // Skip this entire process if foldersToKeepInData is null
+                                {
+                                    if (File.GetAttributes(folderGameData).HasFlag(FileAttributes.Directory))
```

Comment:
If I am understanding correctly, this is to check if the folder specified by the `foreach` is a valid directory?

Reply:
Maybe adding some comments to explain the logic could be helpful here and there

Reply:
this is because limitation of Delete function that can only delete either a file or folder, but not both at the same time, so it has to be in separate function (File.Delete at L560)